api/net/inet.hpp

```diff
@@ -277,16 +281,16 @@ namespace net {
 
     /* Automatic configuration of ipv6 address for inet */
     void autoconf_v6(int retries = 0, slaac_timeout_func = nullptr,
-            IP6::addr alternate_addr = IP6::ADDR_ANY);
+            ip6::Addr alternate_addr = IP6::ADDR_ANY);
 
-    bool is_configured() const
```

Comment:
Many of these functions can still be `const` if you refer to the member instead of the non const `ip_obj()` func (same goes for `ndp()` I guess?).